From 6eb8e98066fb0a9efc1205937885d78677fe393a Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 02:15:51 -0400
Subject: [pci] Convert pci_dev_flags definitions to bit shifts

Message-id: <20140905021551.32574.30681.stgit@gir.stowe>
Patchwork-id: 91361
O-Subject: [RHEL7.1 PATCH 02/17] PCI: Convert pci_dev_flags definitions to bit shifts
Bugzilla: 1136537
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1136537
Upstream Status: 6b121592f8a3fd2bd0de128637b76a0d0864d993

commit 6b121592f8a3fd2bd0de128637b76a0d0864d993
Author: Alex Williamson <alex.williamson@redhat.com>
Date:   Thu May 22 17:07:36 2014 -0600

    PCI: Convert pci_dev_flags definitions to bit shifts

    Convert the pci_dev_flags definitions from decimal constants to bit shifts.
    We're only a few entries away from where using the decimal value becomes
    cumbersome.  No functional change.

    Tested-by: George Spelvin <linux@horizon.com>
    Tested-by: Pat Erley <pat-lkml@erley.org>
    Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/pci.h b/include/linux/pci.h
index a806fb6..58e9665 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -164,13 +164,13 @@ enum pci_dev_flags {
  /* INTX_DISABLE in PCI_COMMAND register disables MSI
   * generation too.
   */
- PCI_DEV_FLAGS_MSI_INTX_DISABLE_BUG = (__force pci_dev_flags_t) 1,
+ PCI_DEV_FLAGS_MSI_INTX_DISABLE_BUG = (__force pci_dev_flags_t) (1 << 0),
  /* Device configuration is irrevocably lost if disabled into D3 */
- PCI_DEV_FLAGS_NO_D3 = (__force pci_dev_flags_t) 2,
+ PCI_DEV_FLAGS_NO_D3 = (__force pci_dev_flags_t) (1 << 1),
  /* Provide indication device is assigned by a Virtual Machine Manager */
- PCI_DEV_FLAGS_ASSIGNED = (__force pci_dev_flags_t) 4,
+ PCI_DEV_FLAGS_ASSIGNED = (__force pci_dev_flags_t) (1 << 2),
  /* Flag for quirk use to store if quirk-specific ACS is enabled */
- PCI_DEV_FLAGS_ACS_ENABLED_QUIRK = (__force pci_dev_flags_t) 8,
+ PCI_DEV_FLAGS_ACS_ENABLED_QUIRK = (__force pci_dev_flags_t) (1 << 3),
 };
 
 enum pci_irq_reroute_variant {
-- 
1.7.1